sdk/clientcore/core/src/main/java/io/clientcore/core/observability/AttributeKey.java

```diff
@@ -0,0 +1,167 @@
+    private final static ReflectiveInvoker GET_TYPE_INVOKER;
+    private final static ReflectiveInvoker CREATE_STRING_KEY_INVOKER;
+    private final static ReflectiveInvoker CREATE_BOOLEAN_KEY_INVOKER;
+    private final static ReflectiveInvoker CREATE_LONG_KEY_INVOKER;
+    private final static ReflectiveInvoker CREATE_DOUBLE_KEY_INVOKER;
+
+    private volatile Object otelAttributeKey;
+    static {
+        Class<?> otelAttributeKeyClass;
+
+        ReflectiveInvoker getKeyInvoker;
+        ReflectiveInvoker getTypeInvoker;
+        ReflectiveInvoker createStringKeyInvoker;
+        ReflectiveInvoker createBooleanKeyInvoker;
+        ReflectiveInvoker createLongKeyInvoker;
+        ReflectiveInvoker createDoubleKeyInvoker;
+
+        try {
+            otelAttributeKeyClass = Class.forName("io.opentelemetry.api.common.AttributeKey", true, Span.class.getClassLoader());
+
+            getKeyInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("getKey"));
+
+            getTypeInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("getType"));
+
+            createStringKeyInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("stringKey", String.class));
+
+            createBooleanKeyInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("booleanKey", String.class));
+
+            createLongKeyInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("longKey", String.class));
+
+            createDoubleKeyInvoker = ReflectionUtils.getMethodInvoker(otelAttributeKeyClass,
+                otelAttributeKeyClass.getMethod("doubleKey", String.class));
+
+        } catch (Exception e) {
+            otelAttributeKeyClass = null;
+
+            getKeyInvoker = null;
+            getTypeInvoker = null;
+            createStringKeyInvoker = null;
+            createBooleanKeyInvoker = null;
+            createLongKeyInvoker = null;
+            createDoubleKeyInvoker = null;
+            // TODO
+        }
+
+        OTEL_ATTRIBUTE_KEY_CLASS = otelAttributeKeyClass;
+
+        GET_KEY_INVOKER = getKeyInvoker;
+        GET_TYPE_INVOKER = getTypeInvoker;
+
+        CREATE_STRING_KEY_INVOKER = createStringKeyInvoker;
+        CREATE_BOOLEAN_KEY_INVOKER = createBooleanKeyInvoker;
+        CREATE_LONG_KEY_INVOKER = createLongKeyInvoker;
+        CREATE_DOUBLE_KEY_INVOKER = createDoubleKeyInvoker;
+    }
+
+    AttributeKey(Object otelAttributeKey) {
+        this.otelAttributeKey = otelAttributeKey;
+    }
+
+    String getKey() {
+        if (otelAttributeKey == null || GET_KEY_INVOKER == null) {
+            return null;
+        }
+
+        try {
+            return (String) GET_KEY_INVOKER.invokeWithArguments(otelAttributeKey);
```

Comment:
It seems a lot of these methods don't have checked exceptions. I wonder if it'd be worth looking into using `LambdaMetafactory` where these reflective handlers could be represented by interfaces such as `Supplier<String>`, `Function<String, Object>`, etc, so every call doesn't need a try-catch block.

Also, is this try-catching even the right design? Or is this just stubs for the time being?

Reply:
The underlying methods are not supposed to throw at all.

Problems in observability should not affect how application works, so the goal is to catch anything caused by the potential version mismatch or reflection bugs, swallow it and turn tracing off since our expectations about the underlying lib API turned out to be wrong.